src/main/java/org/opentripplanner/routing/alertpatch/AlertPatch.java

```diff
@@ -131,79 +133,40 @@ public void apply(Graph graph) {
-            agency = this.agency != null ? agencies.get(this.agency) : null;
-        }
-        Route route = this.route != null ? graph.index.routeForId.get(this.route) : null;
-        Stop stop = this.stop != null ? graph.index.stopForId.get(this.stop) : null;
-        Trip trip = this.trip != null ? graph.index.tripForId.get(this.trip) : null;
-
-        if (route != null || trip != null || agency != null) {
-            Collection<TripPattern> tripPatterns = null;
-
-            if(trip != null) {
-                tripPatterns = new LinkedList<TripPattern>();
-                TripPattern tripPattern = graph.index.patternForTrip.get(trip);
-                if(tripPattern != null) {
-                    tripPatterns.add(tripPattern);
-                }
-            } else if (route != null) {
-               tripPatterns = graph.index.patternsForRoute.get(route);
-            } else {
-                // Find patterns for the feed.
-                tripPatterns = graph.index.patternsForFeedId.get(feedId);
-            }
-
-            if (tripPatterns != null) {
-                for (TripPattern tripPattern : tripPatterns) {
-                    if (direction != null && !direction.equals(tripPattern.getDirection())) {
-                        continue;
-                    }
-                    if (directionId != -1 && directionId != tripPattern.directionId) {
-                        continue;
-                    }
-                    for (int i = 0; i < tripPattern.stopPattern.stops.length; i++) {
-                        if (stop == null || stop.equals(tripPattern.stopPattern.stops[i])) {
-                            graph.removeAlertPatch(tripPattern.boardEdges[i], this);
-                            graph.removeAlertPatch(tripPattern.alightEdges[i], this);
-                        }
-                    }
-                }
-            }
-        } else if (stop != null) {
-            TransitStop transitStop = graph.index.stopVertexForStop.get(stop);
+    public void apply(Graph graph) {
+        applyOrRemove(graph, (Edge e) -> graph.addAlertPatch(e, this));
+    }
 
-            for (Edge edge : transitStop.getOutgoing()) {
-                if (edge instanceof PreBoardEdge) {
-                    graph.removeAlertPatch(edge, this);
-                    break;
-                }
-            }
+    public void remove(Graph graph) {
+        applyOrRemove(graph, (Edge e) -> graph.removeAlertPatch(e, this));
+    }
 
-            for (Edge edge : transitStop.getIncoming()) {
-                if (edge instanceof PreAlightEdge) {
-                    graph.removeAlertPatch(edge, this);
-                    break;
-                }
-            }
-        }
+    /**
+     * Returns true if the trip pattern would NOT be affected by this alert patch.
+     */
+    private boolean isNotApplicableTripPattern(TripPattern tripPattern, Agency alertAgency) {
```

Comment:
Non-blocking Nit (NBN?): maybe rename method to `tripPatternIsUnaffectedByAlert`. Just a touch more readable.

Reply:
I actually originally named the variable similar to that, but decided to prefix with `is` because it seemed like it might follow some kind of naming convention. And in fact there is something I can reference to: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members?redirectedfrom=MSDN

Reply:
How about negating the function and rename to `appliesToTripPattern`?